src/sys/petsclib.cxx

```diff
@@ -47,5 +77,42 @@ void PetscLib::cleanup() {
   count = 0; // ensure that finalise is not called again later
 }
 
+void PetscLib::setPetscOptions(Options& options, std::string pass_options_prefix) {
```

Comment:
It doesn't look like we need a copy, so pass `pass_options_prefix` by `const&`.

Minor point, but I think you can also drop `pass_` from the name too -- it's not clear what that signifies here.

Reply:
I wanted to give the method argument a different name from the class-member-variable `options_prefix`. Maybe the argument could just use a slightly different name, like `prefix`, or do you think it's better to use the same name for both?

Reply:
Ah, I see. Best not to shadow the member variable.